test/performance/scripts/create-outgoing-payments.js

```diff
@@ -158,3 +161,28 @@ export default function (data) {
 
   request(createOutgoingPaymentPayload)
 }
+
+export function handleSummary(data) {
+  const requestsPerSecond = data.metrics.http_reqs.values.rate
+  const iterationsPerSecond = data.metrics.iterations.values.rate
+  const failedRequests = data.metrics.http_req_failed.values.passes
+  const failureRate = data.metrics.http_req_failed.values.rate
+  const requests = data.metrics.http_reqs.values.count
+
+  const summaryText = `
+  **Test Configuration**:
+  - VUs: ${options.vus}
+  - Duration: ${options.duration}
+
+  **Test Metrics**:
+  - Requests/s: ${requestsPerSecond.toFixed(2)}
+  - Iterations/s: ${iterationsPerSecond.toFixed(2)}
+  - Failed Requests: ${failureRate.toFixed(2)}% (${failedRequests} of ${requests})
+    `
+
+  return {
+    // Preserve standard output w/ textSummary
+    stdout: textSummary(data, { enableColors: false }),
+    'k6-test-summary.txt': summaryText // saves to file
+  }
+}
```

Comment:
this is how we're saving the results for use in the comment. the `stdout` will preserve the normal output and the k6-test-summary.txt saves the details we want to report.

Reply:
We could also save a summary HTML report via:

`import { htmlReport } from 'https://raw.githubusercontent.com/benc-uk/k6-reporter/main/dist/bundle.js';`

```typescript
// Report handling
/**
 * Report Generation
 * Handles the generation of test result reports
 * @param {Object} data - Test result data from K6
 * @returns {Object} Report configuration object
 */
export function handleSummary(data) {
    return {
        'reports/summary.html': htmlReport(data)
    };
}
```